src/gnssro/gnssaro_netcdf2ioda.py

```diff
@@ -0,0 +1,494 @@
+    obs_data[('partialBendingAngle', 'MetaData')] = np.array(ifile['Opt_bend_ang'][:], dtype=ioda_float_type)
+
+    return obs_data
+
+
+def get_dateTime(ifile):
+
+    # do the hokey time structure to time structure
+    year = ifile.attrs['year'][0]
+    month = ifile.attrs['month'][0]
+    day = ifile.attrs['day'][0]
+    hour = ifile.attrs['hour'][0]
+    minute = ifile.attrs['minute'][0]
+    second = ifile.attrs['second'][0]  # non-integer value
+    second = round(second)
+
+    # get string date, translate to a datetime object, then offset from epoch
+    dtg = ("%4i-%.2i-%.2iT%.2i:%.2i:%.2iZ" % (year, month, day, hour, minute, second))
+    this_datetime = datetime.strptime(dtg, "%Y-%m-%dT%H:%M:%SZ")
+    time_offset = round((this_datetime - epoch).total_seconds())
+    dateTime = np.int64(time_offset)
+
+    return dateTime
+
+
+def def_meta_data():
+
+    meta_data_keys = {
+        # "qualityFlags": 'radioOccultationDataQualityFlags',
+        "latitude": 'latitude',
+        "longitude": 'longitude',
+        "impactParameterRO": 'impactParameterRO',
+        "impactHeightRO": 'impactHeightRO',
+        "height": 'geometricAltitude',
+        "sensorAzimuthAngle": 'bearingOrAzimuth',
+        "sequenceNumber": 'recordNumber',
+        "satelliteConstellationRO": 'satelliteClassification',
+        "satelliteTransmitterId": 'platformTransmitterIdNumber',
+        "aircraftIdentifier": 'aircraftIdentifier',
+        "aircraftAROInstrument": 'aircraftAROInstrument',
+        "aircraftTailNumber": 'aircraftAROTailNumber',
+        "aircraftAROAntenna": 'aircraftAROSAntenna',
+        "satelliteAscendingFlag": 'originalOccultationAscendingDescendingFlag',
+        "dataProviderOrigin": 'centre',
+        "geoidUndulation": 'geoidUndulation',
+        "earthRadiusCurvature": 'earthLocalRadiusOfCurvature',
+        "geopotentialHeight": 'geopotentialHeight',
+        "partialBendingAngle": 'partialBendingAngle',
+    }
+
+    return meta_data_keys
+
+
+def def_meta_types():
+
+    meta_data_types = {
+        "latitude": "float",
+        "longitude": "float",
+        "dateTime": "long",
+        'impactParameterRO': 'float',
+        'impactHeightRO': 'float',
+        'height': 'float',
+        "sensorAzimuthAngle": 'float',
+        "sequenceNumber": 'integer',
+        "satelliteConstellationRO": 'integer',
+        "satelliteTransmitterId": 'integer',
+        "aircraftIdentifier": 'integer',
+        "aircraftAROInstrument": 'integer',
+        "aircraftTailNumber": 'integer',
+        "aircraftAROAntenna": 'integer',
+        "satelliteAscendingFlag": 'integer',
+        "dataProviderOrigin": 'integer',
+        "geoidUndulation": 'float',
+        "earthRadiusCurvature": 'float',
+        "geopotentialHeight": 'float',
+        "partialBendingAngle": 'float',
+        # "qualityFlags": 'integer',
+    }
+
+    return meta_data_types
+
+
+def concat_obs_dict(obs_data, append_obs_data):
```

Comment:
This looks like a copy of `concat_obs_dict` in `src/lib-python/def_jedi_utils.py `with the addition of a few more data types when appending with fill values. If I've got this right, please consider updating `concat_obs_dict` in `src/lib/lib-python/def_jedi_utils.py` and using that version. This would help with sharing common code. I think there are other scripts using `concat_obs_dict`.

Reply:
Hi @srherbener, thanks for your comment! You are right, I took a look and the functions are very similar, I didn't realize that before, just used what was already in `gnssro_bufr2ioda.py`. I will update `concat_obs_dict` in `src/lib-python/def_jedi_utils.py` in a commit. I will see how that changes for other scripts that function.

Reply:
@ibanos90 I did not get to this